src/js/integrations/devicecontext.ts

```diff
@@ -47,7 +48,14 @@ export class DeviceContext implements Integration {
         event.user = nativeUser;
       }
 
-      const nativeContext = native.context;
+      let nativeContext = native.context;
+      if (AppState.currentState !== 'unknown') {
+        nativeContext = nativeContext || {};
+        nativeContext.app = {
+          ...nativeContext.app,
+          in_foreground: AppState.currentState === 'active',
```

Comment:
Looking at https://reactnative.dev/docs/appstate#app-states
There are edge cases for Android and iOS, wondering if we should do something about it.

Reply:
Based on this:
https://developer.android.com/guide/components/activities/activity-lifecycle#coordinating-activities
https://github.com/getsentry/sentry-java/blob/fe30606689aa4c20db711bb91bfd369169ee18f5/sentry-android-core/src/main/java/io/sentry/android/core/LifecycleWatcher.java#L105

The `sentry-java` impl behaves the same as `active` in RN.